tests/layer/test_hinsage.py

```diff
@@ -573,15 +573,21 @@ def test_hinsage_from_generator():
     # neighbours.
 
     def transform_neighbours(neighs, dim):
-        return np.expand_dims(
-            neighs.reshape(1, dim, int(neighs.shape[1] / dim), neighs.shape[2]).sum(
-                axis=-1
+        return tf.reduce_mean(
+            tf.expand_dims(
+                tf.reduce_sum(
+                    tf.reshape(
+                        neighs, (1, dim, int(neighs.shape[1] / dim), neighs.shape[2])
```

Comment:
This is some serious [Hadouken indentation](http://i.imgur.com/BtjZedW.jpg)!

Maybe:

```python
new_shape = (1, dim ,...)
summed = tf.reduce_sum(tf.reshape(neighs, new_shape)))
return tf.reduce_mean(tf.expand_dims(summed, -1), axis=2)
```

Reply:
haahaha i hadn't heard of that before! that some impressive nesting